x/src/cargo.rs

```diff
@@ -129,6 +157,14 @@ impl Cargo {
             self.inner.arg("--").args(&self.pass_through_args);
         }
 
+        // once all the arguments are added to the command we can log it.
+        if log {
+            self.env_additions
+                .iter()
+                .for_each(|t| info!("Env {:?}: {:?}", t.0, t.1));
+            info!("Executing: {:?}", &self.inner);
+        }
+        self.inner.envs(self.env_additions.to_vec());
```

Comment:
Change this to the following to avoid an extra alloc:
```suggestion
        self.inner.envs(&self.env_additions);
```